tests/integration/suite/conftest.py

```diff
@@ -375,6 +375,26 @@ def clean():
     return _cleanup
 
 
+@pytest.fixture
+def remove_k8s_resource(admin_mc, request):
+    """Remove a k8s resource after a test finishes even if the test fails."""
+    def _cleanup(resource, plural_kind):
```

Comment:
Instead of passing in plural_kind it can be build off the object with `resource['kind'].lower()+'s'`

Reply:
Talked to @vincent99 about this before. Problem is this is not perfect, some things may require intervention. Quick example, some things ending with c require an s some don't. I went ahead added logic to just attach an s if the name ends with s or z and made it overridable.